### PR TITLE
Dilpreet | Test | Pending Voucher Related

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/86cvfh0fa

### CODE REVIEW

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 1]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
-                        <div class="d-flex align-items-center column-gap1">
-                            <div class="custom-datepicker-wrapper">
-                                <div class="custom-datepicker">
-                                    <input type="text" name="selectedDateRange"
-                                        class="giddh-datepicker date-range-picker" />
+                    <div class="vouchers-wrapper pt-3 px-4 overflow-visible">
+                        <div class="invoice-header d-flex align-items-center justify-content-between mr-b15">
+                            <div class="d-flex align-items-center column-gap1">
+                                <dropdown-field
+                                    *ngIf="activeModule === 'credit note' || activeModule === 'debit note'"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Use Enum

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: Add label = voucher type

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 2]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                            name="selectedDateRange"
+                                            [value]="selectedDateRangeUi ? selectedDateRangeUi : ''"
+                                            class="giddh-datepicker date-range-picker border-radius-2"
+                                        />
+                                    </div>
                                 </div>
+                                <button
+                                    mat-stroked-button
+                                    color="primary"
+                                    *ngIf="!isCompany"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: optimse and remove hidden

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 3]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                    [hidden]="selectedPendingVouchers?.length === 0"
+                                    (click)="generateBulkInvoice(false)"
+                                >
+                                    {{ localeData?.generate_multiple }}
+                                </button>
+                                <button
+                                    mat-stroked-button
+                                    color="primary"
+                                    *ngIf="!isCompany"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: optimse and remove hidden

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 4]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                <ng-container matColumnDef="date">
+                                    <th mat-header-cell *matHeaderCellDef>
+                                        {{ commonLocaleData?.app_date }}
+                                    </th>
+                                    <td mat-cell *matCellDef="let element">
+                                        {{ element?.entryDate }}
+                                    </td>
+                                </ng-container>
+                                <ng-container matColumnDef="particular">
+                                    <th mat-header-cell *matHeaderCellDef>Particular</th>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 5]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                            </p>
+                                        </ng-container>
+                                    </td>
+                                </ng-container>
+                                <ng-container matColumnDef="account">
+                                    <th mat-header-cell *matHeaderCellDef>{{ commonLocaleData?.app_account }}</th>
+                                    <td mat-cell *matCellDef="let element" class="no-dbl-click-select">
+                                        <div
+                                            class="cursor-pointer"
+                                            (dblclick)="!isCompany ? (element.showAccountDetails = true) : false"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]:
```suggestion
                                            (dblclick)="!isCompany  && element.showAccountDetails = true"
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 6]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                                            >{{ commonLocaleData?.app_generate_invoice }}</a
+                                                        >
+                                                    </li>
+                                                </ul>
+                                            </mat-card-content>
+                                        </mat-card>
+                                    </td>
+                                </ng-container>
+                                <ng-container matColumnDef="total">
+                                    <th mat-header-cell *matHeaderCellDef class="text-right">Total</th>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 7]
```diff
@@ -1171,116 +1171,234 @@ <h3>{{ element?.customerName | titlecase }}</h3>
...
+                                                [amount]="element.total.amount"
+                                                [currencySymbol]="element.accountCurrencySymbol"
+                                                [currencyCode]="activeCompany?.baseCurrency"
+                                            >
+                                            </amount-field>
+                                        </span>
+                                    </td>
+                                </ng-container>
+                                <ng-container matColumnDef="description">
+                                    <th mat-header-cell *matHeaderCellDef>Description</th>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 8]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
                 </div>
             </mat-tab>
             <!-- payment-content-end -->
+
+               <!-- pending-content-start -->
+            <mat-tab [label]="localeData?.tabs?.pending" *ngIf="activeTabGroup === 3 || activeTabGroup === 4">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:2902]:
```suggestion
            <mat-tab [label]="localeData?.tabs?.pending" *ngIf="(activeModule === 'pending') && (activeTabGroup === 3 || activeTabGroup === 4)">
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 9]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
...
+                <div class="pd-t15" *ngIf="activeModule === 'pending'">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:2903]:
```suggestion
                <div class="pd-t15">
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 10]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
...
+                                            name="selectedDateRange"
+                                            [value]="selectedDateRangeUi ? selectedDateRangeUi : ''"
+                                            class="giddh-datepicker date-range-picker border-radius-2"
+                                        />
+                                    </div>
+                                </div>
+                                <button
+                                    mat-stroked-button
+                                    color="primary"
+                                    *ngIf="!isCompany"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:2920]: optimise and remove hidden

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 11]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
...
+                                    [hidden]="selectedPendingVouchers?.length === 0"
+                                    (click)="generateBulkInvoice(false)"
+                                >
+                                    {{ localeData?.generate_multiple }}
+                                </button>
+                                <button
+                                    mat-stroked-button
+                                    color="primary"
+                                    *ngIf="!isCompany"
+                                    [hidden]="selectedPendingVouchers?.length === 0"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:2930]: optimise and remove hidden

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 12]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
...
+                                    (click)="generateBulkInvoice(true)"
+                                >
+                                    {{ localeData?.generate_compound }}
+                                </button>
+                                <!-- <button
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Remove comment

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 13]
```diff
@@ -2775,6 +2893,231 @@ <h3 class="fonr-16">Kriti - 28Sep</h3>
...
+                                                            >{{ commonLocaleData?.app_generate_invoice }}</a
+                                                        >
+                                                    </li> -->
+                                                </ul>
+                                            </mat-card-content>
+                                        </mat-card>
+                                    </td>
+                                </ng-container>
+                                <ng-container matColumnDef="total">
+                                    <th mat-header-cell *matHeaderCellDef class="text-right">Total</th>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: Check translation in overall page

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 14]
```diff
@@ -299,6 +301,39 @@ export class VoucherListComponent implements OnInit, OnDestroy {
...
+    /** selected profile currency symbol */
+    public baseCurrencySymbol: string = '';
+    /** selected profile currency type */
+    public baseCurrency: string = '';
+    /** True if custom date selected */
+    public customDateSelected: boolean = false;
+    /** Stores the voucher API version of company */
+    private voucherApiVersion: 1 | 2;
+    /** Instance of modal */
+    public modalDialogRef: any;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts]: remove

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 15]
```diff
@@ -299,6 +301,39 @@ export class VoucherListComponent implements OnInit, OnDestroy {
     };
     /** Holds current route query parameters */
     public queryParams: any = {};
+    /** True if today selected */
+    public todaySelected: boolean = false;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts:305]: Check all the new variables are in use or not

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 16]
```diff
@@ -617,6 +662,83 @@ export class VoucherListComponent implements OnInit, OnDestroy {
...
             }
         });
+
+
+        this.componentStore.universalPendingDate$.pipe(takeUntil(this.destroyed$)).subscribe(dateObj => {
+            if (dateObj) {
+                this.universalDate = cloneDeep(dateObj);
+
+                setTimeout(() => {
+                    this.store.pipe(select(state => state.session.todaySelected), take(1)).subscribe(response => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts]: Use this form component store

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 17]
```diff
@@ -617,6 +662,83 @@ export class VoucherListComponent implements OnInit, OnDestroy {
                 this.allVouchersSelected = false;
             }
         });
+
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts:662]: remove extra line

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 18]
```diff
@@ -617,6 +662,83 @@ export class VoucherListComponent implements OnInit, OnDestroy {
...
+                this.isGetAllRequestInProcess$ = of(false);
+                if (this.todaySelected) {
+                    this.ledgerSearchRequest.dateRange = [response.fromDate, response.toDate];
+                    this.ledgerSearchRequest.from = response.fromDate;
+                    this.ledgerSearchRequest.to = response.toDate;
+
+                    this.selectedDateRange = { startDate: dayjs(response.fromDate, GIDDH_DATE_FORMAT), endDate: dayjs(response.toDate, GIDDH_DATE_FORMAT) };
+                    this.selectedDateRangeUi = dayjs(response.fromDate, GIDDH_DATE_FORMAT).format(GIDDH_NEW_DATE_FORMAT_UI) + " - " + dayjs(response.toDate, GIDDH_DATE_FORMAT).format(GIDDH_NEW_DATE_FORMAT_UI);
+                }
+
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts:717]: remove line

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 19]
```diff
@@ -2039,4 +2221,249 @@ export class VoucherListComponent implements OnInit, OnDestroy {
...
+            // Replace placeholders in the tooltip template
+            const currencyConversion = this.localeData?.currency_conversion
+                ?.replace("[BASE_CURRENCY]", this.baseCurrency)
+                ?.replace("[AMOUNT]", grandTotalAmountForCompany.toLocaleString())
+                ?.replace("[CONVERSION_RATE]", grandTotalConversionRate.toString());
+
+            // Assign tooltip text to the item
+            item.totalTooltipText = currencyConversion || '';
+
+        } catch (error) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts]: What is use of this

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 20]
```diff
@@ -220,7 +220,16 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
     public universalDate$: Observable<any> = this.select(this.store.select(state => state.session.applicationDate), (response) => response);
     public createEwayBill$: Observable<any> = this.select(this.store.select(state => state.receipt.voucher), (response) => response);
     public sessionUserEmail$: Observable<any> = this.select(this.store.select(state => state.session.user), (response) => response);
-    
+    public pendingVoucherList$: Observable<any> = this.select(this.store.select(state => state.invoice.ledgers), (response) => response);
+    public isBulkInvoiceGenerated$: Observable<any> = this.select(this.store.select(state => state.invoice.isBulkInvoiceGenerated), (response) => response);
+    public isBulkInvoiceGeneratedWithoutErr$: Observable<any> = this.select(this.store.select(state => state.invoice.isBulkInvoiceGeneratedWithoutErrors), (response) => response);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts]:
```suggestion
    public isBulkInvoiceGeneratedWithoutError$: Observable<any> = this.select(this.store.select(state => state.invoice.isBulkInvoiceGeneratedWithoutErrors), (response) => response);
```

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 21]
```diff
@@ -220,7 +220,16 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
     public universalDate$: Observable<any> = this.select(this.store.select(state => state.session.applicationDate), (response) => response);
     public createEwayBill$: Observable<any> = this.select(this.store.select(state => state.receipt.voucher), (response) => response);
     public sessionUserEmail$: Observable<any> = this.select(this.store.select(state => state.session.user), (response) => response);
-    
+    public pendingVoucherList$: Observable<any> = this.select(this.store.select(state => state.invoice.ledgers), (response) => response);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts:223]: Check all the variables are in used or not

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 1 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 2 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 4 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 5 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 6 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 7 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html]: DONE

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 8 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html:2902]: Not required

File: apps/web-giddh/src/app/vouchers/list/list.component.html (hunk 9 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.html:2903]: Not required

File: apps/web-giddh/src/app/vouchers/list/list.component.ts (hunk 15 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/list/list.component.ts:305]: we are using this

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts (hunk 21 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts:223]: remove unused

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**